ops/testing.py

```diff
@@ -1016,6 +1016,10 @@ def get_pebble(self, socket_path: str):
             self._pebble_clients[socket_path] = client
         return client
 
+    def planned_unit_count(self):
+        items = list(self._meta.relations.items())
+        count = len([relname for relname, _ in items if self._meta.relations[relname].role.is_peer()])
```

Comment:
This doesn't quite make sense to me. It seems to be counting the number of peer *relations* rather than the number of units in a peer relation.

I'm pretty sure we're missing a test_testing.py test that creates a charm with a peer relation, calls 'add_relation_unit(peerid, X)' and then sees that planned_unit_count changes accordingly.

note that if we *really* want it, planned_unit_count should be decoupled from the actual number of units in the relation, as the model there for 'add_relation_unit' is that it is a *joined* unit, and planned_* is very much about the units which should exist but don't yet.
That likely means Harness needs some sort of 'add_planned_unit' of some sort.

Reply:
I think that you might be right. I'm still ingesting all of what the testing harness does. At the very least, I should test my assumptions by adding some units in the planned unit count test, and verify that it changes appropriately.